### ISSUE TITLE
Inline visualization use-case

### ISSUE THREAD

**Issue by danny-waite** [zeit/hyper#2092]
Been tracking Hyper since it first appeared, great project, kudos.  I've had a use case/plugin in mind where the user gets visual responses inline from commands that are executed.   I guess it would be kind of a mashup between Hyper and Vorpal but with the ability to output HTML/CSS blocks.  Something like, `chart currency USD-GBP` and the output is displayed directly below.

I've looked through the many plugins but I can't find something doing similar, also watched @mathisonian great talk which gave me some hints but I'm still scratching my head as to the best way forward.  Any hints/suggestions would be most appreciated.

**Comment by mofux** [comment]
@danny-waite I don't think this is possible without writing the whole terminal rendering from scratch and towards this specific use-case. Anyway, there already is something that comes very close to what you are proposing: https://github.com/railsware/upterm

**Comment by mathisonian** [comment]
The reason why I structured https://github.com/mathisonian/hyperchart the way that I did (with a window in the corner rather than displayed directly inline) is because hyper doesn't really control the terminal output rendering. 

Without a big replacement of that tech, using a dedicated "rich output" area seemed like the most reasonable thing to do. I tried to abstract some of that logic here https://github.com/mathisonian/hyperterm-window

**Comment by danny-waite** [comment]
great feedback guys, I'll check out upterm, thanks for the tip @mofux didn't come across this after plenty of digging! 

yeah I'd seen the window abstraction thanks @mathisonian that had given me hope but seemed a way off where I'd wanted to take it. I'll report back here if I have any success.
